app/src/pages/inside/settingsPage/settingsTabs/autoAnalysisTab/analysisForm/analysisForm.jsx

```diff
@@ -0,0 +1,233 @@
+import { ToggleButton } from 'components/buttons/toggleButton';
+import { StrategyBlock } from './strategyBlock/index';
+import { AccuracyFormBlock } from './accuracyFormBlock/index';
+import styles from './analysisForm.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  analysisModeTitle: {
+    id: 'AnalysisForm.analysisModeTitle',
+    defaultMessage: 'Mode of Auto-Analyse Accuracy',
+  },
+  analysisStrictModeTitle: {
+    id: 'AnalysisForm.analysisStrictModeTitle',
+    defaultMessage: 'Strict',
+  },
+  analysisModerateModeTitle: {
+    id: 'AnalysisForm.analysisModerateModeTitle',
+    defaultMessage: 'Moderate',
+  },
+  analysisLightModeTitle: {
+    id: 'AnalysisForm.analysisLightModeTitle',
+    defaultMessage: 'Light',
+  },
+  updateSuccessNotification: {
+    id: 'AnalysisForm.updateSuccessNotification',
+    defaultMessage: 'Project settings were successfully updated',
+  },
+  updateErrorNotification: {
+    id: 'AnalysisForm.updateErrorNotification',
+    defaultMessage: 'Something went wrong',
+  },
+});
+
+const selector = formValueSelector('analysisForm');
+
+const analysisModeConfig = {
+  Strict: {
+    minShouldMatch: 95,
+    minDocFreq: 5,
+    minTermFreq: 7,
+    numberOfLogLines: 2,
+  },
+  Moderate: {
+    minShouldMatch: 80,
+    minDocFreq: 7,
+    minTermFreq: 1,
+    numberOfLogLines: 2,
+  },
+  Light: {
+    minShouldMatch: 60,
+    minDocFreq: 7,
+    minTermFreq: 1,
+    numberOfLogLines: 2,
+  },
+};
+
+const DEFAULT_ANALYSIS_MODE = 'Moderate';
+
+@injectIntl
+@reduxForm({
+  form: 'analysisForm',
+  validate: ({ minShouldMatch, minTermFreq, minDocFreq }) => ({
+    minShouldMatch:
+      (!minShouldMatch || !validate.minShouldMatch(minShouldMatch)) && 'minShouldMatchHint',
+    minTermFreq: (!minTermFreq || !validate.minTermFreq(minTermFreq)) && 'minTermFreqHint',
+    minDocFreq: (!minDocFreq || !validate.minDocFreq(minDocFreq)) && 'minTermFreqHint',
+  }),
+})
+@connect(
+  (state) => ({
+    analyzerConfiguration: projectAnalyzerConfigSelector(state),
+    currentProject: activeProjectSelector(state),
+    formInputsValues: selector(
+      state,
+      'minShouldMatch',
+      'minDocFreq',
+      'minTermFreq',
+      'numberOfLogLines',
+    ),
+  }),
+  { showNotification },
+)
+export class AnalysisForm extends PureComponent {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    analyzerConfiguration: PropTypes.object,
+    currentProject: PropTypes.string,
+    initialize: PropTypes.func,
+    change: PropTypes.func,
+    handleSubmit: PropTypes.func,
+    showNotification: PropTypes.func,
+    formInputsValues: PropTypes.object,
+  };
+
+  static defaultProps = {
+    analyzerConfiguration: {},
+    currentProject: '',
+    initialize: () => {},
+    change: () => {},
+    handleSubmit: () => {},
+    showNotification: () => {},
+    formInputsValues: {},
+  };
+
+  state = {
+    currentTabsValue: DEFAULT_ANALYSIS_MODE,
```

Comment:
Which tabs? This name is not clear. I suggest using something like `autoAnalysisMode` to make it more clear

Reply:
Fixed